src/data/enum.lean

```diff
@@ -0,0 +1,190 @@
+
+import category.monad.basic
+import data.list.basic
+import data.equiv.basic
+import data.finset
+import data.fintype
+
+open finset (hiding singleton)
+
+namespace fin
+
+def enum (n : ℕ) : list (fin n) :=
+(list.range n).attach.map $ λ ⟨i,h⟩, ⟨i,by simpa using h⟩
+
+@[simp] lemma mem_enum {n} (x : fin n) : x ∈ enum n :=
```

Comment:
Then for this one we should add a missing lemma `list.mem_of_fn`.

Reply:
I missed this comment. Yes, that is what is missing. I can add it.